### ISSUE TITLE
POC Migrate IOScheduler to VirtualThreads 

### ISSUE THREAD

**Issue by ChrisKujawa** [camunda/camunda#14643]
**Description**

Blocked by https://github.com/camunda/zeebe/issues/13716

I was recently checking the [most recent changes in Java 21](https://www.youtube.com/watch?v=E8NV68ihJyY), and project loom. Based on the [Java 21 new feature: Virtual Threads #RoadTo21](https://www.youtube.com/watch?v=5E0LU85EnTI) I was thinking it may fit perfectly our IOScheduler which we use for exporters and AsyncSnapshotDirector, where we do IO.

As far as I understand it makes most sense to use the virtual threads if you do heavily IO, but if you work mostly on memory it might not gain any benefit (due to the overhead of virtual threads). 

I would like to play around with this a bit and try it on the IO Scheduler, so that each submitted ActorJob is actually a virtual thread, instead of using the ActorScheduler mechanics.

I expect that we can increase the exporter throughput since when waiting on IO in a virtual thread the thread is "parked" and the underlying physical thread is released (and continued when IO is done). My expectation is that in this case, another partition can continue with exporting, etc. Would be interesting to play around with this.

**Comment by npepinpe** [comment]
You can check https://github.com/camunda/zeebe/compare/np-virtual-actor?expand=1 where I played around with that. I guess you mean ever ActorTask is a virtual thread, and not job? :thinking: Or really every job?

**Comment by Zelldon** [comment]
Thanks @npepinpe 👍🏼 I wanted to try first every job.

**Comment by Zelldon** [comment]
It's likely that we need to make some changes in the elasticexporter I guess (regarding concurrency 🤔 maybe ), but I guess we will find out.

**Comment by megglos** [comment]
ZDP-Triage:
- virtual threads are cheaper than real OS threads
- another advantage, virtual threads pause on io and leave the os thread utlization to another virtual thread
- would be worth trying it for jobs first

**Comment by marcosgvieira** [comment]
### 🧭 Triage Assessment

| Field | Value |
|------|-------|
| **Work type** | TASK_TOIL_NON_FEATURE |
| **Impact** | ⚠️ Mid |
| **When** | ℹ️ Later |
| **Confidence** | 0.96 |

---

### 📝 Rationale

Exploratory PoC/refactoring for IOScheduler using virtual threads; no current product defect is reported. The affected Zeebe engine/execution area belongs to Core Features. 

---

🤖 _This issue was automatically triaged by an AI-based system as part of the Core Features triage workflow._

✋ **Want a human review?**
If you believe this assessment is incorrect or needs adjustment, feel free to update the details or request a manual triage by adding the **`triage:manual`** label.
